closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeXDHKeyAgreement.java

```diff
@@ -0,0 +1,330 @@
+
+        if (!(key instanceof XECPrivateKey)) {
+            throw new InvalidKeyException
+            ("Unsupported key type");
+        }
+        XECPrivateKey = (XECPrivateKey) key;
+        XECParameters xecParams = XECParameters.get(
+            InvalidKeyException::new, XECPrivateKey.getParams());
+        checkLockedParams(InvalidKeyException::new, xecParams);
+
+        this.ops = new XECOperations(xecParams);
+        this.privateKey = XECPrivateKey.getScalar().orElseThrow(
+            () -> new InvalidKeyException("No private key value")
+        );
+        secret = null;
+
+        if (!NativeCrypto.isAllowedAndLoaded()) {
+            if (nativeCryptTrace) {
+                System.err.println("OpenSSL library not loaded. Using Java crypto implementation to generate secret.");
+            }
+            initializeJavaImplementation();
+        } else {
+            /* Assign ID used by OpenSSL for different curves */
+            int curveType;
+            if (ops.getParameters().getName().contains("X25519")) {
+                curveType = NativeCrypto.X25519;
+            } else {
+                curveType = NativeCrypto.X448;
+            }
+
+            if ((curveType == NativeCrypto.X448)
+            && (NativeCrypto.getVersionIfAvailable() < NativeCrypto.OPENSSL_VERSION_3_0_0)
+            ) {
+                if (nativeCryptTrace) {
+                    System.err.println("OpenSSL version too old for X448 key agreement (<3.x), using Java crypto implementation.");
+                }
+                initializeJavaImplementation();
+            }
+        }
+    }
+
+    @Override
+    protected Key engineDoPhase(Key key, boolean lastPhase)
+            throws InvalidKeyException, IllegalStateException {
+        byte[] computedSecret;
+        if (this.javaImplementation != null) {
+            computedSecret = javaImplComputeSecret(key, lastPhase);
+        } else {
+            if (this.privateKey == null) {
+                throw new IllegalStateException("Not initialized");
+            }
+            if (this.secret != null) {
+                throw new IllegalStateException("Phase already executed");
+            }
+            if (!lastPhase) {
+                throw new IllegalStateException
+                    ("Only two party agreement supported, lastPhase must be true");
+            }
+
+            if (!(key instanceof XDHPublicKeyImpl)) {
+                throw new InvalidKeyException
+                    ("Unsupported key type");
+            }
+
+            XDHPublicKeyImpl publicKey = (XDHPublicKeyImpl) key;
+
+            // Ensure public key parameters are compatible with private key
+            XECParameters xecParams = XECParameters.get(InvalidKeyException::new,
+                publicKey.getParams());
+            if (!ops.getParameters().oidEquals(xecParams)) {
+                throw new InvalidKeyException(
+                "Public key parameters are not compatible with private key.");
```

Comment:
Is there any preference on the folding for that, or should I just pick one and consistently change it across the code?

Reply:
It appears you started with `XDHKeyAgreement.java` which itself is not consistent.

In OMR and OpenJ9, the preference would be to put the opening parenthesis on the same line as the function (or constructor) name, and that should probably be the preference here (see [CodingStandard.md](https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#function-parameter-list-does-not-fit-on-one-line)).

In my opinion, line splitting seems to have been done rather eagerly. For example, I don't think the `throw` statement on lines 94-96 need be split, it could be
```java
                throw new InvalidKeyException("Incorrect private key parameters");
```
which is only 82 characters. OMR's coding standard doesn't offer guidance for a maximum line length, but I don't think 100 would be excessive.

Reply:
I avoided line splitting when character count is less than 100, and I starter a new line with the parameter otherwise.